pre_commit/schema.py

```diff
@@ -206,6 +209,14 @@ def __eq__(self, other):
         return other is not MISSING and other != self.val
 
 
+class NotIn(object):
+    def __init__(self, values):
```

Comment:
I'd probably do `__init__(self, *values)` just to save some parens (other classes in this module do similarly).

p4: bonus points if it's a namedtuple (so we get free `repr`):

```python
class NotIn(collections.namedtuple('NotIn', ('values',))):
    def __new__(cls, *values):
        return super(NotIn, cls).__new__(cls, values=values)
    def __eq__(...):
        ...
```